### PR TITLE
Add Chatty and feedbackd

### PR DESCRIPTION
**Pull request by masipcat**
###### Motivation for this change

Use this amazing application from Librem

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by masipcat** [comment]
Some screenshots:


image

image

**Comment by masipcat** [comment]
Right now Chatty is built without any libpurple plugin and only works as a xmpp client. I'm not sure if we want to do something similar to pidgin-with-plugins. What do you think?

**Comment by masipcat** [comment]
When this PR is merged the package `calls` can be update to latest version cc/ @craigem @lheckemann

**Comment by masipcat** [comment]
Do you want to review this PR @jtojnar @flokli ? :)

File: pkgs/applications/networking/instant-messengers/chatty/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    sqlite
+  ];
+
+  postPatch = ''
+    chmod +x build-aux/meson/postinstall.py
+    patchShebangs build-aux/meson/postinstall.py
+  '';
+
+  meta = with stdenv.lib; {
+    description = "XMPP and SMS messaging via libpurple and Modemmanager";
```
Comment (jtojnar) [inline pkgs/applications/networking/instant-messengers/chatty/default.nix:54]: ModemManager

**Comment by masipcat** [comment]
@jtojnar  Fixed :)

**Comment by jtojnar** [comment]
Yeah, agree that we will need `chatty-with-plugins`. Also likely a module for feedbackd.

**Comment by masipcat** [comment]
> Also likely a module for feedbackd.

What do you mean?

**Comment by jtojnar** [comment]
Since it is a daemon, it probably requires installing by setting `systemd.packages`, `services.dbus.packages` NixOS options or something like that. NixOS module could handle that.

**Comment by masipcat** [comment]
Added chatty-with-plugins. Successfully tested the following ones:

```
plugins = [ pkgs.pidgin-carbons pkgs.purple-lurch pkgs.telegram-purple ];
```

**Comment by jtojnar** [comment]
Looks like `feedbackd` does not seem to compile gsettings schemas. And also it does have a dbus service so it will not work without being installed globally (e.g. using a module).

**Comment by masipcat** [comment]
> Looks like `feedbackd` does not seem to compile gsettings schemas. And also it does have a dbus service so it will not work without being installed globally (e.g. using a module).

Working on a module + fixing gsettings!

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -70,6 +70,7 @@ mapAliases ({
   casperjs = throw "casperjs has been removed, it was abandoned by upstream and broken.";
   catfish = xfce.catfish; # added 2019-12-22
   cgmanager = throw "cgmanager was deprecated by lxc and therefore removed from nixpkgs."; # added 2020-06-05
+  chatty-with-plugins = chatty; # added 2020-08-16
```
Comment (jtojnar) [inline pkgs/top-level/aliases.nix:73]: Why add an legacy alias when this never existed?

Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix:73]: Please remove this. Aliases are for package renames/moves/etc.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 94820` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chatty</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>feedbackd</li>
  </ul>
</details>

chatty:
```
src/meson.build:60:0: ERROR: Dependency "libhandy-0.0" not found, tried pkgconfig
```

**Comment by dotlambda** [comment]
@masipcat Are you still interested in working on this?

**Comment by jtojnar** [comment]
Feedbackd is now packaged. Jordi did not seem to respond on their other Librem related PR so feel free to revive chatty in a new PR.

**Comment by dotlambda** [comment]
closing in favor of https://github.com/NixOS/nixpkgs/pull/122373

**Pull request closed without merging**